packages/common/address.ts

```diff
@@ -20,16 +20,40 @@ export interface Address {
 }
 
 export interface AddressInputParts {
-  street: string
+  street: string // including street number, e.g. 100 S Biscayne Blvd
   city: string
   state: string
   postcode: string
   unit?: string
 }
 
+/** Removes unnecessary and trailing spacing */
+const trim = (s: string) => s.replace(/( +)/g, ' ').trim()
+
+/** Returns a formatted unit, using pound (#) only when necessary */
+export const formatUnit = (unit: string) => {
+  // Users might use pounds erroneously, let this function decide whether
+  // to use it or not.
+  //
+  // We also remove unnecessary spacing, for example the double spacing
+  // in 'BLDG  3'
+  const normalized = trim(unit.replace(/#/g, ''))
+
+  // Searchs for any abbreviation that don't require pounds
+  //
+  // The list of officially approved abbreviations can be found here: https://pe.usps.com/text/pub28/28apc_003.htm#ep538629
+  // But we'll remove pounds if any word is found in the Unit field, since users
+  // might not use the officially approved abbreviations or just commit a typo
+  // while filling this field, e.g. APY 303.
+  const dontUsePound = /( +|^)([a-z]+)( +|$)/i
+  return !normalized.toUpperCase().match(dontUsePound)
+    ? trim(`# ${normalized}`)
+    : normalized
```

Comment:
Did this because I don't know the side effects of having cases like `# APY 303`, I don't think limiting/validating users input of address.unit at `<AddressInput/>` would be a good idea, since some people might not know the list of officially approved address.

Reply:
I'm not too worried about these cases.  People are not going to write things like "APT" in that small text field.